jwst/resample/resample.py

```diff
@@ -73,6 +73,7 @@ def __init__(self, input_models, output=None, single=False, blendheaders=True,
         if output is not None and '.fits' not in str(output):
             self.output_dir = output
             self.output_filename = None
+        self.output_suffix = 'i2d'
```

Comment:
Both here and in `resample_spec`, the `.output_suffix` attribute seems redundant with `.suffix`, which is set by all the pipelines e.g. [here](https://github.com/spacetelescope/jwst/blob/5bce1b404578df64cb0ac7dac460571265afc0ce/jwst/pipeline/calwebb_spec3.py#L87).  It looks like `.suffix` is supposed to be handled by `stpipe.Step` with [this function](https://github.com/spacetelescope/stpipe/blob/7a3fea69a014fc766b80ff1330ce9fc80db38558/src/stpipe/step.py#L1390).  Can this existing functionality be leveraged to solve the problem, or is it insufficient for some reason and this separate attribute is needed?

Reply:
This is not a class that inherits from stpipe.Step.  We could pass it from the calling steps, but we'd still need to store it similarly in the class.  In this case, it's used only to construct the intermediate filenames when resample is called as many-to-many for outlier detection, so I'm not sure it's worth it to add more arguments to pass it down from the calling steps.

Reply:
This is maybe another argument for using temp files to fix the collisions in JP-3717, though.

Reply:
ok that makes sense.  Minor nit-pick but it might be clearer to name these `outlier_i2d` and `outlier_s2d` and/or name the attribute `intermediate_suffix`.  Other than that this PR looks great to me

Reply:
Good idea, I'll make that change.